templates/batch/resources/usr/local/bin/liferay_batch_entrypoint.sh

```diff
@@ -54,10 +54,53 @@ function main {
 		exit 1
 	fi
 
-	find /opt/liferay/batch -type f -name "*.batch-engine-data.json" -print0 |
+	local site_initializer_json="/opt/liferay/site-initializer/site-initializer.json"
+
+	if [ -e "${site_initializer_json}" ]
+	then
+		echo "Processing Site Initializer: ${site_initializer_json}"
+		echo ""
+
+		local href="/o/headless-site/v1.0/sites/by-external-reference-code/"
+
+		echo "HREF: ${href}"
+
+		local site=$(jq -r 'del(.["dxp.lxc.liferay.com.virtualInstanceId"])' ${site_initializer_json})
+
+		echo "Site: ${site}"
+
+		local external_reference_code=$(jq -r ".externalReferenceCode" <<< "${site}")
+
+		local site_initializer_zip="/opt/liferay/site-initializer/site-initializer.zip"
+
+		local put_response=$(\
+			curl \
+				-H "Accept: application/json" \
+				-H "Authorization: Bearer ${oauth2_access_token}" \
+				-H "Content-Type: multipart/form-data" \
+				-X PUT \
+				-F "file=@${site_initializer_zip};type=application/zip" \
+				-F "site=${site}" \
+				-s \
+				${curl_options} \
+				"${lxc_dxp_server_protocol}://${lxc_dxp_main_domain}${href}${external_reference_code}" \
+			| jq -r ".")
+
+		echo "PUT Response: ${put_response}"
+		echo ""
+
+		if [ ! -n "${put_response}" ]
+		then
+			echo "Received invalid PUT response."
+
+			exit 1
+		fi
+	fi
```

Comment:
So the `site` var would also include all the optional extra properties here, right? Would we want to also exclude `externalReferenceCode` from the `site` var as well (like you did with `dxp.lxc.liferay.com.virtualInstanceId`)?

Reply:
No, externalReferenceCode is a field of the site model.

Reply:
Interesting, so it's part of both the path and the payload?

Reply:
My point is that there's no need to remove it from the json since it's part of the Site resource. However, in reality we are using the json as the "transport" for the information, the ERC is used in the API call so it needs to be delivered to the logic and then extracted and pass in the API call. Beyond that, it's ignored in the provided resource. That means it will always be duplicate anyway making the need to remove it unnecessary.

Reply:
@drewbrokke I made a change that eliminated that quick